pkg/ccl/backupccl/restore_online.go

```diff
@@ -91,15 +180,30 @@ func sendAddRemoteSSTs(
 
 	fromSystemTenant := isFromSystemTenant(dataToRestore.getTenantRekeys())
 
-	restoreWorkers := int(onlineRestoreLinkWorkers.Get(&execCtx.ExecCfg().Settings.SV))
-	for i := 0; i < restoreWorkers; i++ {
-		grp.GoCtx(sendAddRemoteSSTWorker(execCtx, restoreSpanEntriesCh, requestFinishedCh, *kr, fromSystemTenant, &approxRows, &approxDataSize))
+	if err := execCtx.ExecCfg().JobRegistry.CheckPausepoint("restore.before_split"); err != nil {
+		return 0, 0, err
 	}
 
-	if err := grp.Wait(); err != nil {
-		return 0, 0, errors.Wrap(err, "failed to generate and send remote file spans")
+	if err := job.NoTxn().RunningStatus(ctx, "Splitting and distributing spans"); err != nil {
+		return 0, 0, err
 	}
-	return approxRows, approxDataSize, nil
+
+	if err := splitAndScatter(ctx, execCtx, genSpan, *kr, fromSystemTenant, targetRangeSize); err != nil {
+		return 0, 0, errors.Wrap(err, "failed to split and scatter spans")
+	}
+
+	if err := execCtx.ExecCfg().JobRegistry.CheckPausepoint("restore.before_link"); err != nil {
+		return 0, 0, err
+	}
+
+	if err := job.NoTxn().RunningStatus(ctx, ""); err != nil {
```

Comment:
did you mean to add an ingestion status here?

Reply:
no, I meant to clear the split status set previously

Reply:
we set running status when we're doing something notable, but we're just doing the body of the job here so I don't think we need a special status displayed.